assets/js/atomic/utils/get-block-map.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import { getRegisteredInnerBlocks } from '@woocommerce/blocks-registry';
+
+/**
+ * Internal dependencies
+ */
+import {
+	ProductTitle,
+	ProductPrice,
+	ProductButton,
+	ProductImage,
+	ProductRating,
+	ProductSummary,
+	ProductSaleBadge,
+} from '../components/product';
+
+/**
+ * Map blocks names to components.
+ *
+ * @param {string} blockName Name of the parent block. Used to get extension children.
+ */
+export const getBlockMap = ( blockName ) => ( {
+	'woocommerce/product-price': ProductPrice,
+	'woocommerce/product-image': ProductImage,
+	'woocommerce/product-title': ProductTitle,
+	'woocommerce/product-rating': ProductRating,
+	'woocommerce/product-button': ProductButton,
+	'woocommerce/product-summary': ProductSummary,
+	'woocommerce/product-sale-badge': ProductSaleBadge,
+	...getRegisteredInnerBlocks( blockName ),
```

Comment:
I see this is not new .. I was not aware of this. Do we need a hint/reminder/process to update this if we ever change those blocks? I guess the main component and the block name are both quite stable. This feels like it could potentially be generated from block config. 

None of this is a blocker, just the first time I've seen this woo innerblocks magic :)

Longer term, some of this might be appropriate for Gutenberg APIs - i.e. other blocks plugins might want to allow innerblock editing in a similar manner to Woo Blocks. (I'm guessing this already has been thought about & discussed somewhat!)

Reply:
Extensibility is a follow up. R/e the map, there is no API, config, or method in Gutenberg to map a block to a component, that's why this has to exist. We're saying if you see this block name, render this component. This is for frontend usage where we use dynamic components rather than saved content like regular gutenberg blocks.